### PR TITLE
[WIP][EUWE]BZ#1510469-add pagination component

### PR DESCRIPTION
**Pull request by AllenBW**
https://bugzilla.redhat.com/show_bug.cgi?id=1510469

A WIP until the [api work relating to subquery_count is backported to euwe](https://github.com/ManageIQ/manageiq/commit/ffe76e106d2a2a56ab26e993764199d0cd9a92cd)

## Don't freak!! Read the following carefully
* "Results" and pagination appear broken in the gif, as soon as subquery_count is available both of this things will work as expected 
* tests fail, but they fail for euwe branch, its an issue that exceeds the scope of this work (which has already ballooned)

### Where we're at
pag

## Important notes
* filtering by chargeback is gone
* filtering by retirement has change to true and false, (titles and functionality limited by ang-pf 3.x), **false** will show all services with retirement scheduled, **true** will show all retired services (gif demonstrates this)
* sorting is done on the client, meaning the user is only sorting by what they see

### CODE REVIEW

**Comment by AllenBW** [comment]
screen shot 2017-11-08 at 1 16 37 pm

Of the above filters, the only one we will have trouble on filtering is the relative cost (as this is calculated client side and then sorted)

**Comment by AllenBW** [comment]
screen shot 2017-11-08 at 1 16 37 pm

Of the above filters, the only one we will have trouble on filtering is the relative cost (as this is calculated client side and then sorted)

**Comment by AllenBW** [comment]
screen shot 2017-11-08 at 1 16 37 pm

Of the above filters, the only one we will have trouble on filtering is the relative cost (as this is calculated client side and then sorted)

**Comment by AllenBW** [comment]
screen shot 2017-11-08 at 1 16 37 pm

Of the above filters, the only one we will have trouble on filtering is the relative cost (as this is calculated client side and then sorted)

**Comment by AllenBW** [comment]
screen shot 2017-11-08 at 1 16 37 pm

Of the above filters, the only one we will have trouble on filtering is the relative cost (as this is calculated client side and then sorted)

File: client/app/resources/collections-api.factory.js [hunk 1]
```diff
@@ -5,7 +5,7 @@
     .factory('CollectionsApi', CollectionsApiFactory);
 
   /** @ngInject */
-  function CollectionsApiFactory($http, API_BASE) {
+  function CollectionsApiFactory($http, API_BASE, lodash) {
```
Comment (chriskacerguis) [inline client/app/resources/collections-api.factory.js]: Can we use lodash on node 0.10?

Comment (AllenBW) [reply client/app/resources/collections-api.factory.js]: Yep!  We already have throughout the code on the euwe branch

**Comment by AllenBW** [comment]
@chriskacerguis looks like https://github.com/ManageIQ/manageiq/pull/16377 made it in, this is g2g!

**Comment by AllenBW** [comment]
Gonna close this till its neeEEEeeeeEEEEEEEE**EEEEEEE**ded

💌 👍 🥇

**Pull request closed without merging**